include/zephyr/init.h

```diff
@@ -173,4 +173,9 @@ void z_sys_init_run_level(int32_t level);
 }
 #endif
 
+/* This should be called by platforms that do not wish to
+ * have mbedtls initialised during kernel startup
+ */
+int mbedtls_init(const struct device *device);
+
```

Comment:
I don't think that is the right place to add it. Maybe would be better a header like `modules/mbedtls/zephyr_mbedtls.h` that you include to the include path when mbedTLS is enabled.

Reply:
Thanks for feedback, let me go noodle in the patch and see what i can come up with.